### PR TITLE
Implement targeting fencing levels by node attribute

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
If a fencing level target is specified as "name=value", then the level will apply to any nodes that have a node attribute in the CIB with that name and value.

There was some discussion of the best symbol to use as the separator, e.g.:
   --target="rack=1" (current approach in this code)
   --target="rack#1"
   --target="rack%1"

Not an option are any characters that are legal in node names, or regex special characters.

While --target="rack:1" looks nice, it would require distinguishing from an IPv6 address. Similarly, --target="rack==1" might be clearer, but would make detection and name/value splitting more complicated. You can see the simplicity of the code now vs having to do any significant string processing.

**Comment by beekhof** [comment]
I think we need to support rack:1
IPv6 addresses need more than one colon so differentiating is only one additional call to `strchr()`

File: cts/CIB.py [hunk 1]
```diff
@@ -30,7 +28,7 @@ def __setitem__(self, key, value):
         if value:
             self.kwargs[key] = value
         else:
-            self.values.append(key)
+            self.kwargs.pop(key, None)
```
Comment (beekhof) [inline cts/CIB.py]: Doesn't `pop()` remove values?

Comment (beekhof) [inline cts/CIB.py]: Oh, oh course. NM

**Comment by kgaillot** [comment]
Do you think we should support both '=' and ':' or just ':'? I'm leaning to just ':' b/c any complexity we add must be implemented in pcs/crmsh as well.

BTW there could be some confusion, since target is normally a hostname, people might think "a:b" is a "host:port" specification. I think that's trivial and doesn't stop us from using ':', but we might run into it.

**Comment by beekhof** [comment]
I think both.  Both shells already have to deal with similar logic for pcmk_host_map

**Pull request merged**